code/player.dm

```diff
@@ -1,27 +1,61 @@
-//set stuff that has to persist between connections here.
+/// for client variables and stuff that has to persist between connections
 /datum/player
-	var
-		key
-		client/client
-
-		mentor = 0
-		see_mentor_pms = 1
-
-		shamecubed = 0
-
-		list/cooldowns = list()
+	/// the ckey of the client object that this datum is attached to
+	var/key
+	/// the client object that this datum is attached to
+	var/client/client
+	/// are they a mentor?
+	var/mentor = 0
+	/// do we want to see mentor pms?
+	var/see_mentor_pms = 1
+	/// to make sure that they cant escape being shamecubed by just reconnecting
+	var/shamecubed = 0
+	/// how many rounds theyve declared ready and joined, null with to differentiate between not set and no participation
+	var/rounds_participated = null
+	/// how many rounds theyve joined to at least the lobby in, null to differentiate between not set and not seen
+	var/rounds_seen = null
+	/// a list of cooldowns that has to persist between connections
+	var/list/cooldowns = null
 
+	/// sets up vars and caches player stats
 	New(key)
 		src.key = key
 		src.tag = "player-[ckey(key)]"
+		src.cooldowns = list()
+
+		if (mentors.Find(ckey(src.key)))
+			src.mentor = 1
+
+		if (src.key) //just a safety check!
+			src.cache_round_stats()
```

Comment:
I'm not sure if it's a good idea to do this here or to leave it to the first get_round_stats call. I'd probably just remove it and let the data get loaded when / if anyone asks for them.

Reply:
im keeping this in because i want to avoid potential hangups when assigning jobs, especially if more jobs get time gated in the future.